addons/html_builder/static/src/website_sale/products_item_option_plugin.js

```diff
@@ -0,0 +1,429 @@
+    if (created.length > 0) {
+      proms.push(
+        this.services.orm
+          .create(
+            "product.ribbon",
+            created.map((ribbon) => {
+              ribbon = Object.assign({}, ribbon);
+              this.originalRibbons[ribbon.id] = ribbon;
+              delete ribbon.id;
+              return ribbon;
+            })
+          )
+          .then((ids) => (createdRibbonIds = ids))
+      );
+    }
+
+    for (let ribbon of modified) {
+      const ribbonData = {
+        name: ribbon.name,
+        bg_color: ribbon.bg_color,
+        text_color: ribbon.text_color,
+        position: ribbon.position,
+      };
+      proms.push(
+        this.services.orm.write("product.ribbon", [ribbon.id], ribbonData)
+      );
+      this.originalRibbons[ribbon.id] = Object.assign({}, ribbon);
+    }
+
+    if (deletedIds.length > 0) {
+      proms.push(this.services.orm.unlink("product.ribbon", deletedIds));
+    }
+
+    await Promise.all(proms);
+
+    const localToServer = Object.assign(
+      this.ribbonsObject,
+      Object.fromEntries(
+        created.map((ribbon, index) => [
+          ribbon.id,
+          { ...this.ribbonsObject[ribbon.id], id: createdRibbonIds[index] },
+        ])
+      ),
+      {
+        false: {
+          id: "",
+        },
+      }
+    );
+
+    // Building the final template to ribbon-id map
+    const finalTemplateRibbons = this.productTemplatesRibbons.reduce(
+      (acc, { templateId, ribbonId }) => {
+        acc[templateId] = ribbonId;
+        return acc;
+      },
+      {}
+    );
+    // Inverting the relationship so that we have all templates that have the same ribbon to reduce RPCs
+    const ribbonTemplates = {};
+    for (const templateId in finalTemplateRibbons) {
+      const ribbonId = finalTemplateRibbons[templateId];
+      if (!ribbonTemplates[ribbonId]) {
+        ribbonTemplates[ribbonId] = [];
+      }
+      ribbonTemplates[ribbonId].push(parseInt(templateId));
+    }
+
+    const promises = [];
+    for (const ribbonId in ribbonTemplates) {
+      const templateIds = ribbonTemplates[ribbonId];
+      const parsedId = parseInt(ribbonId);
+      const validRibbonId = currentIds.includes(parsedId) ? ribbonId : false;
+      promises.push(
+        this.services.orm.write("product.template", templateIds, {
+          website_ribbon_id: localToServer[validRibbonId]?.id || false,
+        })
+      );
+    }
+
+    return Promise.all(promises);
+  }
+
+  /**
+   * Deletes a ribbon.
+   *
+   */
+  _deleteRibbon(editingElement) {
+    const ribbonId = parseInt(editingElement.dataset.ribbonId);
+    if (this.ribbonsObject[ribbonId]) {
+      this.deletedRibbonClasses += ` ${
+        this.ribbonPositionClasses[this.ribbonsObject[ribbonId].position]
+      }`;
+
+      const ribbonIndex = this.ribbons.indexOf(
+        this.ribbons.find((ribbon) => ribbon.id === ribbonId)
+      );
+      if (ribbonIndex >= 0) this.ribbons.splice(ribbonIndex, 1);
```

Comment:
we usually never do if statement in one line. it is annoying when we want to add code, or when we need to add a debugger
```js
      if (ribbonIndex >= 0) {
        this.ribbons.splice(ribbonIndex, 1);
   }
```